libs/platform/core/src/services/hydrate/default-hydrate.service.ts

```diff
@@ -0,0 +1,120 @@
+import { inject, INJECTOR, OnDestroy } from '@spryker-oryx/di';
+import {
+  hydratableAttribute,
+  HYDRATE_ON_DEMAND,
+  PatchableLitElement,
+  rootInjectable,
+} from '@spryker-oryx/utilities';
+import { LitElement } from 'lit';
+import { Subscription } from 'rxjs';
+import { AppRef, ComponentsPlugin } from '../../orchestration';
+import { HydrateInitializer, HydrateService } from './hydrate.service';
+
+export class DefaultHydrateService implements HydrateService, OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected componentsPlugin = inject(AppRef).findPlugin(ComponentsPlugin),
+    protected root = rootInjectable.get(),
+    protected injector = inject(INJECTOR)
+  ) {
+    this.initialize();
+  }
+
+  protected initialize(): void {
+    const initializers = this.injector.inject(HydrateInitializer, null);
+
+    if (!initializers) {
+      return;
+    }
+
+    for (const initializer of initializers) {
+      this.subscription.add(initializer(this, this.injector).subscribe());
```

Comment:
I'd improve this one a bit. We should consider either:
- creating generic app initializers (then logic would stay the same)
- or create specific hydration initializers, we can react to here (so they won't have to do it in a initializer)

Also we sohuld not have to pass injector, as using factory and `inject` is more proper way to inject stuff.